openerp/addons/base/tests/test_ir_filters.py

```diff
@@ -275,3 +275,24 @@ def test_update_default_filter(self):
             dict(name='a', user_id=False, is_default=False, domain='[]', context='{}'),
             dict(name='b', user_id=False, is_default=True, domain='[]', context=context_value),
         ])
+
+from openerp.tests.common import TransactionCase
+
+class TestReadGroup(TransactionCase):
+    """Test function read_group with groupby on a many2one field to a model
+    (in test, "user_id" to "res.users") which is ordered by a inherited not stored field (in
+    test, "name" inherited from "res.partners").
+    """
+
+    def setUp(self):
+        super(TestReadGroup, self).setUp()
+        self.ir_filters_model = self.env['ir.filters']
+        self.res_partner_model = self.env['res.partner']
+        self.res_users_model = self.env['res.users']
+
+    def test_read_group_1(self):
+        filter_a = self.ir_filters_model.create(dict(name="Filter_A", model_id="ir.filters"))
+        filter_b = self.ir_filters_model.create(dict(name="Filter_B", model_id="ir.filters"))
+        filter_b.write(dict(user_id=False))
+        res = self.ir_filters_model.read_group([], ['name', 'user_id'], ['user_id'])
+        self.assertEquals(2, len(res), "Two groups should be returned: Administrator and Undefined")
```

Comment:
or maybe something like

    self.assertTrue(any(val['user_id'] == False for val in res))